What’s up in Vietnam.md

```diff
@@ -1,62 +1,63 @@
 
-## Why is it problematic.
-There are words such as „incompatible type systems” even in the definition, and all incompatibilities are a potential problem. My first question which appeared after getting into the subject was „Why the hell we are using incompatible systems? Why is it the first (and often the only one known) solution to solve the problem with persisting data?”
+## Why is it problematic?
+ „Incompatible type systems” are mentioned already in the definition, and all incompatibilities are potential problems. My first question that popped into my head after getting into the subject was: „Why the hell we are using incompatible systems? Why is it the first (and often the only one known) solution to solve the problem with persisting data?”
 
-> If your project really does not need any relational data features, then ORM will work perfectly for you, but then you have a different problem: you're using the wrong data store. […] On the other hand, if your data is relational, then your object mapping will eventually break down.  
+> If your project really does not need any relational data features, then ORM will work perfectly for you, but then you have a different problem: you're using the wrong data store. […] On the other hand, if your data is relational, then your object mapping will eventually break down.
 _Laurie Voss_
 
 ### Let’s characterize these two worlds
 
-> Object systems are typically characterized by four basic components: identity, state, behavior and encapsulation.  
+> Object systems are typically characterized by four basic components: identity, state, behavior and encapsulation.
 _Ted Neward_
 
-Summing up the above thought, we can say that every object is described by its identity independently of its state. Every object has a collection of operations that can be invoked to manipulate the state of itself or connected objects, and (most importantly) has encapsulated access to itself and effectively, the ability to inheritance and polymorphism.
+Elaborating on the above though we can say that every object is described by its identity independently of its state. Every object has a collection of operations that can be invoked to manipulate the state of itself or connected objects, and (most importantly) has encapsulated access to itself and effectively, possibility of inheritance and polymorphism.
 
-> The representation of your object in memory depends what you intend to do with it, and context-sensitive representation is not a feature of OO design.  
+> The representation of your object in memory depends  on what you intend to do with it, and context-sensitive representation is not a feature of OO design.

```

Comment:
I understand you and I can see that mistake, but it's a quote.

Reply:
Then the quote had a mistake or was in different context

Reply:
The whole context:
```
But if your data is not relational, then you are adding a huge and unnecessary overhead by using SQL in the first place and then compounding the problem by adding a further abstraction layer on top of that.

On the the other hand, if your data is relational, then your object mapping will eventually break down. SQL is about relational algebra: the output of SQL is not an object but an answer to a question. If your object "is" an instance of X and "has" a number of Y, and each of Y "belongs to" a Z, what is the correct representation in memory of your object? Is it merely the properties of X, or should it include all the Ys, and/or all the Zs? If you get only the properties of X, when do you run the query to fetch the Ys? And do you want one or all of them? In reality, it depends: that's what I mean when I say SQL is the answer to a question. *The representation of your object in memory depends what you intend to do with it, and context-sensitive representation is not a feature of OO design.* Relations are not objects; objects are not relations.
```

Reply:
Changed to `The representation of your object in memory depends [sic] what you intend to do with it,`